docs/quick-start-guide.md

```diff
@@ -197,10 +197,22 @@ A note about `metrics-labels`: In `Prometheus`, every unique combination of key-
 Additionally, these metrics are best-effort for the current operator run and will be reset on an operator restart. Also some of these metrics are generated by listening to pod state updates for the driver/executors
 and deleting the pods outside the operator might lead to incorrect metric values for some of these metrics.
 
-## Driver UI Access and Ingress
+## Driver UI Access, Service annotations and Ingress
 
-The operator, by default, makes the Spark UI accessible by creating a service of type `NodePort` which exposes the UI via the node running the driver.
-The operator also supports creating an Ingress for the UI. This can be turned on by setting the `ingress-url-format` command-line flag. The `ingress-url-format` should be a template like `{{$appName}}.ingress.cluster.com` and the operator will replace the `{{$appName}}` with the appropriate appName.
+The operator, by default, makes the Spark UI accessible by creating a service of type `NodePort` which exposes the UI via the node running the driver. 
+
+If you are using API gateway that depends on service annotations to expose services (e.g. [Ambassador](https://www.getambassador.io/)), you can configure a templated service annotations by setting the `spark-ui-service-annotations` command-line flag. The `spark-ui-service-annotations` should be a JSON string and supports the following mustache variables:
+- `{{$appName}}`
+- `{{$serviceName}}`
+- `{{$namespace}}`
+
+The following example demonstrate how to config a route to the spark UI pod based on the `appName` with ambassador:
+```bash
```

Comment:
Can we show the example in a different form that uses part of the deployment yaml?